test/unit/rules/no-autofocus-attribute-test.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+
+generateRuleTests({
+  name: 'no-autofocus-attribute',
+
+  config: true,
+
+  good: ['<input />'],
+
+  bad: [
+    {
+      template: '<input autofocus />',
+      verifyResults(results) {
+        expect(results).toMatchInlineSnapshot(`
+          Array [
+            Object {
+              "column": 7,
+              "endColumn": 16,
+              "endLine": 1,
+              "filePath": "layout.hbs",
+              "isFixable": true,
+              "line": 1,
+              "message": "No autofocus attribute allowed, as it reduces accessibility by moving users to an element without warning and context",
+              "rule": "no-autofocus-attribute",
+              "severity": 2,
+              "source": "autofocus",
+            },
+          ]
+        `);
+      },
+    },
+    {
+      template: '<input autofocus="true" />',
+      verifyResults(results) {
+        expect(results).toMatchInlineSnapshot(`
+          Array [
+            Object {
+              "column": 7,
+              "endColumn": 23,
+              "endLine": 1,
+              "filePath": "layout.hbs",
+              "isFixable": true,
+              "line": 1,
+              "message": "No autofocus attribute allowed, as it reduces accessibility by moving users to an element without warning and context",
+              "rule": "no-autofocus-attribute",
+              "severity": 2,
+              "source": "autofocus=\\"true\\"",
+            },
+          ]
+        `);
+      },
+    },
+    {
+      template: '<input autofocus={{false}} />',
```

Comment:
FYI, this won't render the `autofocus` attribute and thus not cause the a11y issue described. Not sure if this matters?